src/activity_handlers.cpp

```diff
@@ -2728,6 +2729,44 @@ void activity_handlers::try_sleep_finish( player_activity *act, player *p )
     act->set_to_null();
 }
 
+void activity_handlers::build_do_turn( player_activity *act, player *p )
+{
+    const std::vector<construction> &list_constructions = get_constructions();
+    const construction &built = list_constructions[act->index];
```

Comment:
this could be changed to `list_constructions[pc->id]` and moved a little farther down this function, then we don't have to set the index of the activity.  I know that it doesn't really matter that much, but its better to send all the information one way instead of two separate ways to avoid (future) bugs.